kevlar/varmap.py

```diff
@@ -40,22 +34,19 @@ def __init__(self, contig, cutout, score=None, cigar=None, strand=1,
         self.contig = contig
         self.cutout = cutout
         self.score = score
-        self.cigar = cigar
         self.strand = strand
         self.matedist = None
+
+        self.tok = AlignmentTokenizer(self.varseq, self.refrseq, cigar)
+        self.cigar = self.tok._cigar
+
         self.vartype = None
-        self.alnmatch = None
-        for pattern, (vartype, rightcheck) in patterns.items():
-            matchobj = re.match(pattern, cigar)
-            if matchobj:
-                if rightcheck:
-                    idx = 6 if vartype == 'snv' else 9
-                    rightmatchlen = int(matchobj.group(idx))
-                    if rightmatchlen > 5:
-                        continue
-                self.alnmatch = matchobj
-                self.vartype = vartype
-                break
+        snvpattern = '^((\d+)([DI]))?(\d+)M((\d+)[DI])?$'
+        indelpattern = '^((\d+)([DI]))?(\d+)M(\d+)([ID])(\d+)M((\d+)[DI])?$'
```

Comment:
Previously the regular expression match objects were used to break down the alignment into its respective blocks. Now the regular expressions are only used to classify variant type, and the CIGAR tokenizer is used for block-by-block access of the aligned sequences.